uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/TCGMemoryOverwriteRequest/BlackBoxTest/TCGMemoryOverwriteRequestBBTestFunction.c

```diff
@@ -0,0 +1,2390 @@
+ *                        test
+ *  @return EFI_SUCCESS   Successfully.
+ *  @return Other value   Something failed.
+ */
+EFI_STATUS
+TCGMemoryOverwriteRequestControlLockSetVariable (
+  IN EFI_STANDARD_TEST_LIBRARY_PROTOCOL       *StandardLib,
+  IN EFI_TEST_RECOVERY_LIBRARY_PROTOCOL       *RecoveryLib,
+  IN RESET_DATA                               *ResetData
+  )
+{
+  EFI_STATUS                           Status;
+  EFI_TEST_ASSERTION                   Result;
+  UINTN                                DataSize;
+  UINT8                                MemoryOverwriteRequestControlLockData;
+  UINT32                               Attributes;
+
+  //
+  // Check Step to see which part of test remains to be executed
+  //
+  switch (ResetData->Step) {
+    case 0:
+      goto INITIAL_RESET;
+    case 1:
+      goto MORLOCK_SET_VARIABLE;
+    default:
+      return EFI_INVALID_PARAMETER;
+  }
+
+INITIAL_RESET:
+  //
+  // Reset MORLOCK variable by performing a cold reset
+  //
+
+  ResetData->Step = 1;
+  ResetData->CheckpointStep = LOCK_SET_VARIABLE_TEST;
+
+  Status = RecoveryLib->WriteResetRecord (
+                  RecoveryLib,
+                  sizeof (RESET_DATA),
+                  (UINT8*)ResetData
+                  );
+  if (EFI_ERROR(Status)) {
+    StandardLib->RecordAssertion (
+                  StandardLib,
+                  EFI_TEST_ASSERTION_FAILED,
+                  gTestGenericFailureGuid,
+                  L"TestRecoveryLib - WriteResetRecord",
+                  L"%a:%d:Status - %r",
+                  __FILE__,
+                  (UINTN)__LINE__,
+                  Status
+                  );
+    return Status;
+  }
+
+  SctPrint (L"System will cold reset after 1 second and test will be resumed after reboot.");
+  Status = gtBS->Stall(1000000);
+  if (EFI_ERROR(Status)) {
+      // Handle stall error
+      SctPrint(L"Error: Failed to stall the system.\n");
+      return Status;
+  }
+  gtRT->ResetSystem (
+                  EfiResetCold,
+                  EFI_SUCCESS,
+                  0,
+                  NULL
+                  );
+  SctPrint(L"Error: Failed to perform a cold reset.\n");
+  return EFI_DEVICE_ERROR;
+
+MORLOCK_SET_VARIABLE:
+  //
+  // reset step count and Update Checkpoint to LOCK_UNLOCKED_STATE_TEST
+  //
+
+  ResetData->Step = 0;
+  ResetData->CheckpointStep = LOCK_UNLOCKED_STATE_TEST;
+
+  Status = RecoveryLib->WriteResetRecord (
+                  RecoveryLib,
+                  sizeof (RESET_DATA),
+                  (UINT8*)ResetData
+                  );
+  if (EFI_ERROR(Status)) {
+    StandardLib->RecordAssertion (
+                  StandardLib,
+                  EFI_TEST_ASSERTION_FAILED,
+                  gTestGenericFailureGuid,
+                  L"TestRecoveryLib - WriteResetRecord",
+                  L"%a:%d:Status - %r",
+                  __FILE__,
+                  (UINTN)__LINE__,
+                  Status
+                  );
+    return Status;
+  }
+

```

Comment:
Should we check if the lock value is MOR_LOCK_DATA_UNLOCKED before running all the tests? If the data value is MOR_LOCK_DATA_LOCKED before running the tests, we'll get the false alarm because we expected that the data remains MOR_LOCK_DATA_UNLOCKED.

Reply:
The lock SHOULD be reset to MOR_LOCK_DATA_UNLOCKED after each cold reset, the test assumes that this functionality is present and working correctly, I guess we can add this check to be sure that this is not an issue, and if it does we report the error.

Reply:
Sounds good. Thanks, @aabhi64.